data/waveform_generator.js

```diff
@@ -0,0 +1,147 @@
+                           [0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 2, 3, 4, 5, 6, 5, 4, 3, 2, 1, 0],
+                           [0, 0, 0, 0, 0, 1, 2, 3, 4, 5, 6, 5, 4, 3, 2, 1, 0, 0, 0, 0, 0]]
+        this.name = "Waveform Generator";
+        this.configure();
+    }
+
+    configure(){
+        if (this.has_noise){
+            this.n_num_features = 40; // number of features when has_noise = true
+        }
+        this.n_features = this.n_num_features;
+        this.feature_names =[];
+        for(let i = 0; i<this.n_features; i++){
+            this.feature_names.push('att_num_' + i);
+        }
+        this.target_names = ['target_0']; 
+        this.target_values = []
+        for(let i = 0; i<this.n_classes; i++){
+            this.target_values.push(i);
+        }
+    }
+
+    has_noise(){
+        /** Retrieve the value of the option: add noise.
+        *
+        * Returns
+        * -------
+        * Boolean
+        *   True if the noise is added.
+        */
+        
+        return (this.has_noise)
+    }
+
+    set_has_noise(has_noise){
+        /** Set the value of the option: add noise.
+        *
+        * Parameters
+        * ----------
+        * has_noise: Boolean
+        *
+        */
+        if (typeof(has_noise) === 'boolean'){
+            this.has_noise = has_noise;
+            this.configure();
+        }
+        else{throw TypeError('has_noise should be boolean, \'' + typeof(has_noise) +  '\' was passed')} //error
+    }
+
+    prepare_for_use(){
+        /**
+        * Should be called before generating the samples.
+        */
+        
+        //self.random_state = check_random_state(self._original_random_state)
+        this.sample_idx = 0
+    }
+
+    next_sample(batch_size = 1){
+        /**  next_sample
+        *
+        * An instance is generated based on the parameters passed. If noise 
+        * is included the total number of features will be 40, if it's not
+        * included there will be 21 attributes. In both cases there is one 
+        * classification task, which chooses one between three labels.
+        * 
+        * After the number of attributes is chosen, the algorithm will randomly
+        * choose one of the hard coded waveforms, as well as random multipliers. 
+        * For each attribute, the actual value generated will be a a combination 
+        * of the hard coded functions, with the multipliers and a random value.
+        * 
+        * Furthermore, if noise is added the features from 21 to 40 will be
+        * replaced with a random normal value.
+        * 
+        * Parameters
+        * ----------
+        * batch_size: int
+        *     The number of samples to return.
+        *     
+        * Returns
+        * -------
+        * tuple or tuple list
+        *     Return a tuple with the features matrix and the labels matrix 
+        *     for the batch_size samples that were requested.
+        */
+
+        let rand = new Utils()
```

Comment:
Let's call this object utils, not rand. so you will use later utils.rand_gauss(...).